PLANET-3182 Configure wpunit suite

kirdia opened the pull request:
Integrate codeception wpunit suite in ci 
Related to this [PR](https://github.com/greenpeace/planet4-base-fork/pull/15)

koyan commented:
Konstantino's IDE now is setup as requested.

comzeradd commented:
We could add an `.editorconfig` file to this repo too, to ensure this. Same to what we have on [other repos](https://github.com/greenpeace/planet4-master-theme/blob/develop/.editorconfig#L11). All modern IDEs support editorconfig.

27Bslash6 commented on Makefile:
```diff
@@ -227,6 +227,7 @@ endif
 
 db/Dockerfile:
 	envsubst < $@.in > $@
+	cp ci/scripts/duplicate-db.sh defaultcontent/duplicate-db.sh
```
This is not the right Makefile rule for this.
Why not place it in the db/ folder so it's already in the Docker build context?

27Bslash6 commented on Makefile:
```diff
@@ -257,8 +258,9 @@ test: test-env-info test-codeception
-	@docker-compose exec php-fpm tests/vendor/bin/codecept run --xml=results.xml --html
+	@docker-compose exec php-fpm bash -c 'cd tests && composer install --prefer-dist --no-progress'
+	@docker-compose exec php-fpm tests/vendor/bin/codecept run wpunit --xml=junit.xml --html --debug
+	@docker-compose exec php-fpm tests/vendor/bin/codecept run acceptance --xml=junit.xml --html
```
Is it worth splitting this rule into three different targets so they can each be run independently? eg, `install-codeception test-codeception-unit test-codeception-acceptance` ?

kirdia replied:
Yes, good suggestion :+1: will split it

27Bslash6 commented on app.env:
```diff
@@ -5,4 +5,4 @@ VIRTUAL_HOST=www.planet4.test
 WP_STATELESS_MEDIA_ENABLED=true
 WP_STATELESS_MEDIA_BUCKET=stateless-test-planet4-dev
-WP_STATELESS_MEDIA_KEY_FILE_PATH=/app/secrets/wp-stateless-media-key.json
+WP_STATELESS_MEDIA_KEY_FILE_PATH=/app/secrets/wp-stateless-media-key.json
```
Please clean up

27Bslash6 commented on ci/scripts/duplicate-db.sh:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+echo "Create planet4 wpunit database"
+mysql -u root --password="$MYSQL_ROOT_PASSWORD" -e "CREATE DATABASE $TEST_WPUNIT_DB_NAME"
```
No trailing newline.

27Bslash6 commented on db/Dockerfile.in:
```diff
@@ -1,2 +1,3 @@
 FROM mysql:5.7
 ADD ${CONTENT_DB} /docker-entrypoint-initdb.d/
+ADD duplicate-db.sh /docker-entrypoint-initdb.d/
```
As per hadolint, if you're not expecting Docker to perform remote pulls or to extract .gz during an ADD command, use COPY: https://github.com/hadolint/hadolint/wiki/DL3020
```
dev/stdin:3 DL3020 Use COPY instead of ADD for files and folders
```

This error should have been picked up if you have run `make init` before committing changes.

kirdia commented:
Rebased on master and squashed commits.
I did not have a good idea where to put this, if you have any suggestions, please share
https://github.com/greenpeace/planet4-docker-compose/pull/19/files#diff-b67911656ef5d18c4ae36cb6741b7965R190

The ci job fails, because it depends on base-fork related branch
https://github.com/greenpeace/planet4-base-fork/pull/15

But both of them can be checked here
https://circleci.com/gh/greenpeace/planet4-master-theme/1950

27Bslash6 commented on Makefile:
```diff
@@ -187,6 +187,7 @@ build : run unzipimages config elastic flush
 .PHONY : run
 run:
 	@$(MAKE) -j init getdefaultcontent db/Dockerfile
+	cp ci/scripts/duplicate-db.sh defaultcontent/duplicate-db.sh
```
You could replace `defaultcontent` with the variable `$(CONTENT_PATH)` and extract to a Makefile rule such as:
```Make
$(CONTENT_PATH)/duplicate-db.sh: $(CONTENT_PATH)
    cp ci/scripts/duplicate-db.sh $@
```

Then set this new rule `$(CONTENT_PATH)/duplicate-db.sh` as a prerequisite for `run` and any other rules that need the file.
